auth.py

```
@@ -105,4 +105,100 @@ def complex_auth_function(user, pass_word, email, phone, address, city, state, z
+                                    if user_data['address'].get('state'):
+                                        return True
+    return False
+
+# Issue 26: Performance - Inefficient list comprehension
+def filter_active_users(all_users):
+    active_users = [user for user in all_users if user['status'] == 'active']
+    return [user for user in active_users if user['last_login'] != None]
+
+# Issue 27: Best Practices - Not using context manager for file operations
+def write_user_log(message):
+    file = open('user_activity.log', 'a')
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)